---
title: "Semi-Supervised Mouse data Deconvolution"
author: "Xiaoyu Lu"
header-includes:
- \usepackage{amsmath}
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
#vignette: > 
#%\VignetteIndexEntry{stationery} 
#%\VignetteEngine{knitr::rmarkdown} 
#%\VignetteEncoding{UTF-8}
---

```{r,include=F}
library(ggplot2)
library(gplots)
```

<style type="text/css">

h1.title {
  font-size: 28px;
  color: Black;
  text-align: left;
}
h1 { /* Header 1 */
  font-size: 24px;
  color: Black;
}

h2 { /* Header 2 */
  font-size: 20px;
  color: Black;
}

h4.author { /* Header 4 - and the author and data headers use this too  */
    font-size: 18px;
  font-family: "Times New Roman", Times, serif;
  color: Black;
  text-align: left;
}
h4.date { /* Header 4 - and the author and data headers use this too  */
  font-size: 18px;
  font-family: "Times New Roman", Times, serif;
  color: Black;
  text-align: left;
}
body {
text-align: justify}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction
Deconvolution of mouse transcriptomic data is challenged by the fact that mouse models carry various genetic and physiological perturbations, making it questionable to assume fixed cell types and cell type marker genes for different dataset scenarios. We developed a **S**emi-**S**upervised **M**ouse data **D**econvolution (SSMD) method to study the mouse tissue microenvironment (TME). SSMD is featured by (i) a novel non-parametric method to discover data set specific cell type signature genes; (ii) a community detection approach for fixing cell types and their marker genes; (iii) a constrained matrix decomposition method to solve cell type relative proportions that is robust to diverse experimental platforms. In summary, SSMD addressed several key challenges in the deconvolution of mouse tissue data, including: (1) varied cell types and marker genes caused by highly divergent genotypic and phenotypic conditions of mouse experiment, (2) diverse experimental platforms of mouse transcriptomics data, (3) small sample size and limited training data source, and (4) capable to estimate the proportion of 35 cell types in blood, inflammatory, central nervous or hematopoietic systems. In silico and experimental validation of SSMD demonstrated its high sensitivity and accuracy in identifying (sub) cell types and predicting cell proportions comparing to state-of-the-arts methods. A user-friendly web server of SSMD are released via https://ssmd.ccbb.iupui.edu. For more detailed discussion, please see our paper https://www.biorxiv.org/content/10.1101/2020.09.22.309278v1.abstract.

<br>

```{r figurename, echo=FALSE, fig.cap="Analysis pipeline of core marker training and deconvolution procedure", out.width = '90%'}
knitr::include_graphics("/Users/xiaoyulu/Documents/GitHub/SSMD/fig/pipeline.png")
```

<br>

# Basic usage of __SSMD__ package

We demonstrate basic usage of SSMD package using simulated bulk tissue RNA-seq data.

```{r}
library(SSMD)
load('example_bulk.rdata')
```

This loads *example_bulk* object which contains the input bulk tissue RNA-seq data matrix $X$ which has 24672 genes and 100 samples. We utilize SSMD to identify cell types and their corresponding proportion in each sample. Predefined tissue environment information are required to be clarified when we run SSMD. There are four options: "Inflammatory", "Central Nervous System", "Hematopoietic System" or "Blood". We choose "Inflammatory" for *example_bulk* because it is from mouse cancer tissue.

<br>

Main function of SSMD package can be used as below example:

```{r , warning=FALSE}
load('SSMD_res.RData')
#Prediction=SSMD(bulk_data=example_bulk,tissue = 'Inflammatory')
```


# Interpretation of SSMD results

## Predicted Proportion


The most desired result is identifiable cell type and the corresponding proportions. 

```{r}
Predicted_Proportion=Prediction$Proportion
```

```{r,echo=FALSE}
set.seed(123)
colnames(Predicted_Proportion)=paste('Sample',c(1:dim(Predicted_Proportion)[2]),sep = ' ')
print(Predicted_Proportion[c(4,8,13,17,20,22,25),1:5])
```

Row names of Predicted_Proportion is identifiable cell type/sub-type and columns represent samples. Each row means the relative proportion in each sample for a specific cell type.

## Marker Genes

Another important part of SSMD result is 

```{r}
Marker_Genes=Prediction$marker_gene
```

```{r,echo=F,warning=FALSE,fig.show='hold'}
print(Marker_Genes[c(4,20,22)])

data11=example_bulk[Marker_Genes[[4]],]
data11_cor=cor(t(data11))

  colors = c(-100:100)/60
  par(cex.main=0.8)
  my_palette <- colorRampPalette(c("firebrick3", 'white',"darkblue"))(n = 200)
  aa=heatmap.2(as.matrix(data11_cor),Rowv=T,Colv =F,scale="row",
            col=my_palette,breaks=colors,density.info="none",dendrogram="none",
            trace="none",cexRow=0.4,cexCol=0.4,key=F) 
```

This includes all gene modules which are used as markers for each predicted cell type.

## Explanation Score

Explanation score (ES) was utilized to evaluate the goodness that each marker geneâ€™s expression is fitted by the predicted cell proportions

```{r}
ES=Prediction$Escore
```

```{r,echo=FALSE}
print(ES[sample(length(ES),16),])
```

## Potential Cell Types

Last part of SSMD result is Potential Modules. Those modules are significant co-expressed modules which can be used to predict possible rare cell types or subtypes. Those rare cell types or subtypes cannot be confidently identified by SSMD. Users can further investigate and decide if it indeed represents a distinct or rare cell type. 

```{r}
Potential_Modules=Prediction$potential_modules
```

```{r,echo=FALSE}
names(Potential_Modules)=paste('Potential Module',c(1:length(Potential_Modules)),sep = ' ')
print(Potential_Modules[1:2])
```




